### ISSUE TITLE
Current State of Decorators

### ISSUE THREAD

**Issue by dcwardell7** [wycats/javascript-decorators#75]
Thanks for all the work that has been put into designing this feature.

I noticed that it was removed from Babel (currently available as a legacy plugin), and that the last commit was made in November.

I am curious what the current state of this feature is from the designer's perspective. Is it still being actively developed? What are the current issues holding delaying it? Are there any opportunities for others to assist?

**Comment by marklundin** [comment]
There's an ECMA meeting being held in a couple of days on the 25th May in Munich. @wycats will be discussing the proposal and will hopefully have some good feedback.

You can see the [agenda here](https://github.com/tc39/agendas/blob/master/2016/05.md) and find the  the [notes](https://github.com/tc39/tc39-notes/tree/master/es7) after.

**Comment by silkentrance** [comment]
@marklundin Might I perhaps point you towards https://github.com/jeffmo/es-class-fields-and-static-properties/issues/38 (https://github.com/jeffmo/es-class-fields-and-static-properties/issues/38#issuecomment-214850134) which will solve existing issues we have when decorating such initialized properties, see https://github.com/jayphelps/core-decorators.js/issues/62#issuecomment-211030001.
Could this be taken into account when discussing the proposal?

Also, I believe, that https://github.com/wycats/javascript-decorators/issues/70 and https://github.com/wycats/javascript-decorators/issues/66 should also be on the radar.

**Comment by marklundin** [comment]
You can see some of the discussion from the previous [meeting in Jan](https://github.com/tc39/tc39-notes/blob/master/es7/2016-01/2016-01-27.md#decorators) when. Lots of interesting points raised.

**Comment by marklundin** [comment]
Updated spec here https://github.com/tc39/proposal-decorators/blob/master/spec.html

**Comment by fdn** [comment]
@marklundin There aren't any notes posted from the most recent meeting yet. Do you know when those will go live? Is there a diff on the spec changes for decorators too?

**Comment by fdn** [comment]
Is there an updated document since the May meeting? Is there a short diff on what changed since the recent meeting for decorators?

**Comment by marklundin** [comment]
There's an open PR with Mays notes here https://github.com/rwaldron/tc39-notes/pull/45/files#diff-2

It's quite the read, but the long and short is it didn't advance to stage 2.

**Comment by fdn** [comment]
@marklundin thanks for the update. bummer.

**Comment by dcwardell7** [comment]
For any interested, the July tc39 meeting took place last week. Decorators are now in stage 2. For all involved, good work. Notes from the meeting are found [here](https://github.com/tc39/tc39-notes/blob/master/es7/2016-07/jul-28.md#9iiic-decorators).

**Comment by silkentrance** [comment]
@quicksilver7 now, IMHO the process is flawed. it should be more open in that the ongoing discussions here are participated by the people who are responsible for making the proposal happen by working out the specification details. as of now, there was no participation whatsoever on behalf of @wycats.
in addition this needs to be made a collaborative attempt in regards to @jeffmo and the specification proposal part that he is working on, namely class properties. The latter, as I see it, are a major concern of the people over at angular. And by finding a solution for that, the decorators should be well on their way.

**Comment by wycats** [comment]
@silkentrance I'm really sorry. The specification and discussion moved to a repo in the tc39 org (https://github.com/tc39/proposal-decorators) and I left this open to allow any open conversations (especially about future features, like parameter and function decorators) to continue. 

If you think it makes sense, I can turn this repo into a redirect or modify the readme to point people to the right place.

**Comment by callumlocke** [comment]
> If you think it makes sense, I can turn this repo into a redirect or modify the readme to point people to the right place.

Yes please! A prominent link at the top of the readme would have prevented me wasting the last hour of my life :)
